### DISCUSSION TITLE
Regarding GUTI registration

### DISCUSSION THREAD

**Discussion by panyogesh** [aligungr/UERANSIM#347] [General]
Any CLI or method to generate GUTI registaration.

**Comment by aligungr** [comment]
@panyogesh 

UE state machine automatically decides what kind of identity it should use such as GUTI, SUPI, IMEI etc. as specified in TS 24.501.

However, periodic and mobility updating registration are *usually* performed with GUTI. You may want to check them.

More on TS 24.501

**Reply by Fangqierin** [reply]
Do we have a way to specify to use GUTI registration instead of automatically choose from them in UERANSIM?

**Reply by aligungr** [reply]
@Fangqierin 

Sorry, no. There is no way to set the identity type (guti/suci etc.) to be used in registration procedure.
I think manually specifying the identity type is very incompatible with the semantics of 3GPP.
Instead, UE automatically makes this decision with a (relatively) complex procedure.

---

**(From TS 24.501, 5.5.1.2.2 and 5.5.1.3.2):**

During initial registration the UE handles the 5GS mobile identity IE in the following order:
1. if the UE holds a valid 5G-GUTI that was previously assigned, over 3GPP access or non-3GPP access, by 
the same PLMN with which the UE is performing the registration, the UE shall indicate the 5G-GUTI in the 5GS 
mobile identity IE; 
2. if the UE holds a valid 5G-GUTI that was previously assigned, over 3GPP access or non-3GPP access, by an 
equivalent PLMN, the UE shall indicate the 5G-GUTI in the 5GS mobile identity IE;
3. if the UE holds a valid 5G-GUTI that was previously assigned, over 3GPP access or non-3GPP, by any other 
PLMN, the UE shall indicate the 5G-GUTI in the 5GS mobile identity IE; 
4. if a SUCI is available the UE shall include the SUCI in the 5GS mobile identity IE; and 
5. if the UE does not hold a valid 5G-GUTI or SUCI, and is initiating the registration procedure for emergency 
services, the PEI shall be included in the 5GS mobile identity IE.
